crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -2549,9 +2547,14 @@ impl<'db> TypeInferenceBuilder<'db> {
         };
 
         let symbol_ty = if except_handler_definition.is_star() {
-            // TODO: we should infer `ExceptionGroup` if `node_ty` is a subtype of `tuple[type[Exception], ...]`
-            // TODO: should be generic with `symbol_ty` as the generic parameter
-            KnownClass::BaseExceptionGroup.to_instance(self.db())
+            let class = if symbol_ty
+                .is_subtype_of(self.db(), KnownClass::Exception.to_instance(self.db()))
```

Comment:
using `is_subtype_of` rather than `is_assignable_to` here because I think we should infer `BaseExceptionGroup` rather than `ExceptionGroup` if `symbol_ty` is `Unknown` or similar